DOCSUP-10293: Documented the optimize_functions_to_subcolumns setting (#25607)
ClickHouse/ClickHouse: docs/en/sql-reference/data-types/map.md

└───────────────────────────────┘
```

> ## Subcolumn Map.keys {#subcolumn-keys}

Reviewer: ```suggestion
## Map.keys and Map.values Subcolumns {#map-subcolumns}
```

> To read all keys of a `Map` you can use the subcolumn `keys`, which doesn't read the whole column.

Reviewer: ```suggestion
To optimize the `Map` column processing, in some cases you can use the `keys` and 'values' subcolumns instead of reading the whole column.
```

**Example**

Query:

``` sql
CREATE TABLE t_map (`a` Map(String, UInt64)) ENGINE = Memory;

INSERT INTO t_map VALUES (map('key1', 1, 'key2', 2, 'key3', 3));

> SELECT a.keys FROM t_map;

Reviewer: ```suggestion
SELECT a.keys, a.values FROM t_map;
```

> SELECT a.keys FROM t_map;

Reviewer: ```suggestion
SELECT a.keys FROM t_map;

SELECT a.values FROM t_map;
```

```

Result:

``` text

> ┌─a.keys─────────────────┐
> │ ['key1','key2','key3'] │
> └────────────────────────┘

Reviewer: ```suggestion
┌─a.keys─────────────────┬─a.values─┐
│ ['key1','key2','key3'] │ [1,2,3]  │
└────────────────────────┴──────────┘

```

> ┌─a.keys─────────────────┐
> │ ['key1','key2','key3'] │
> └────────────────────────┘

Reviewer: ```suggestion
┌─a.keys─────────────────┬─a.values─┐
│ ['key1','key2','key3'] │ [1,2,3]  │
└────────────────────────┴──────────┘
```

> ┌─a.keys─────────────────┐
> │ ['key1','key2','key3'] │
> └────────────────────────┘

Reviewer: ```suggestion
┌─a.keys─────────────────┐
│ ['key1','key2','key3'] │
└────────────────────────┘

┌─a.values─┐
│ [1,2,3]  │
└──────────┘
```

```

> ## Subcolumn Map.values {#subcolumn-keys}
> 
> To read all values of a `Map` you can use the subcolumn `values`, which doesn't read the whole column.
> 
> **Example**
> 
> Query:
> 
> ``` sql
> CREATE TABLE t_map (`a` Map(String, UInt64)) ENGINE = Memory;
> 
> INSERT INTO t_map VALUES (map('key1', 1, 'key2', 2, 'key3', 3))
> 
> SELECT a.values FROM t_map;
> ```
> 
> Result:
> 
> ``` text
> ┌─a.values─┐
> │ [1,2,3]  │
> └──────────┘
> ```
> 

Reviewer: ```suggestion
```
